### ISSUE TITLE
USWDS-Site: Update the heading structure on component pages with multiple variants

### ISSUE THREAD

**Issue by amyleadem** [uswds/uswds-site#2601]
## Summary
Evaluate the header structure for component pages with multiple variant sections, like the [heading page](https://designsystem.digital.gov/components/header/). More details in the comments below:

Header page:
> Thanks for adding the in-page nav here. I see some issues with the heading levels here, and I'm not sure this PR is the place to fix them, but I wanted to call them out. I think the "Component preview," "Component code," "Guidance," etc., headings should be nested under each heading type (e.g., nested under "Basic header"). To me, it is not logical for these to be at the same level as the heading type--they belong under the heading type.
>
> _Originally posted by @sarah-sch in https://github.com/uswds/uswds-site/pull/2595#discussion_r1550211149_

Language selector page:
> Similar to the header nav, I don't think the preview, code, guidance headers, etc. should have the same heading level as the language selector type under which they fall. This guidance should be nested at a lower heading level.
> _Originally posted by @sarah-sch in https://github.com/uswds/uswds-site/pull/2595#discussion_r1550213682_
